src/main/java/pl/nitl/employeeManager/controlers/EmployeeController.java

```diff
@@ -1,50 +1,61 @@
 package pl.nitl.employeeManager.controlers;
 
+import lombok.extern.slf4j.Slf4j;
+import org.mapstruct.factory.Mappers;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 import pl.nitl.employeeManager.exceptions.EmployeeNotFoundException;
+import pl.nitl.employeeManager.mappers.EmployeeMapper;
 import pl.nitl.employeeManager.models.Employee;
 import pl.nitl.employeeManager.respositories.EmployeeRespository;
+import pl.nitl.employeeManager.views.EmployeeDto;
 
 import java.util.List;
 
+@Slf4j
 @RestController
 public class EmployeeController {
     @Autowired
     EmployeeRespository repository;
+    private EmployeeMapper mapper
+            = Mappers.getMapper(EmployeeMapper.class);
 
     @PostMapping("/employee")
-    public Employee addEmployee(@RequestBody Employee employee) {
-        System.out.println("adding new employee " + employee);
-        repository.save(employee);
-        return employee;
+    public Employee addEmployee(@RequestBody EmployeeDto employeeDto) {
+        Employee mappedEmployee = mapper.employeeDtoToEmployee(employeeDto);
+        log.info("adding new employee " + employeeDto + "and mapping it to " + mappedEmployee);
```

Comment:
"adding new employee " + employeeDto + "and mapping it to " + mappedEmployee - taki zapis powoduje dodawanie i tworzenie doatkochych okietow String, w przypadku logowania na poziomie info moze ni ema to tak duzego znaczenia to jesli logowalbys na poziomie debug a te operacje zostalyby wykonane niepotrzebnie.

lepszy jest zapis typu :
log.info("adding new employee  {} and mapping it to {}", employeeDto, mappedEmployee)
lub jesli nie mozesz uzyc powyzszego:
lof.info(String.format("adding new employee  %s and mapping it to %s", employeeDto, mappedEmployee));